GMnetENGINE.gmx/scripts/htme_clientStart.gml

```diff
@@ -37,6 +37,31 @@ if( self.socketOrServer<0 ){
     return false;
 }
 
+// Get server port from list
+if server_port=0 and udphp_provide_server_port
+{
+    var l = global.udphp_downloadlist;
+    if (ds_exists(l,ds_type_list)) 
+    {
+        for (var i=0; i<ds_list_size(l); i+=1)
+        {
+            var entry = l[| 0];
+            var ip = entry[? "ip"];
+            var port_provided = entry[? "data8"];

```

Comment:
Instead of using data8 (which is reserved for the user) we should add a completely new variable. 

This means GATE.PUNCH needs to be changed as well.
I was actually going to throw the current GATE.PUNCH out of the window and redo it in nodejs since nodejs works so much better for this kind of purpose than java. I will also add a new field for that then.